### PR TITLE
Bump GPOPT version to 1.617

### PR DESCRIPTION
**Pull request by xinzweb**
The new version of GPOPT can handle NULL pointer with partition elimination over full outer join queries.

### CODE REVIEW

File: src/test/regress/expected/partition_full_outer_join.out [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   ->  Hash Left Join  (cost=0.01..0.06 rows=2 width=16)
+         Hash Cond: public.s1.d1 = public.s2.d2 AND public.s1.p1 = public.s2.p2
+         ->  Append  (cost=0.00..0.00 rows=1 width=8)
+               ->  Seq Scan on s1_1_prt_2 s1  (cost=0.00..0.00 rows=1 width=8)
+                     Filter: p1 = 1
+         ->  Hash  (cost=0.00..0.00 rows=1 width=8)
+               ->  Append  (cost=0.00..0.00 rows=1 width=8)
+                     ->  Seq Scan on s2_1_prt_2 s2  (cost=0.00..0.00 rows=1 width=8)
+                           Filter: p2 = 1
+ Optimizer status: legacy query optimizer
```
Comment (vraghavan78) [inline src/test/regress/expected/partition_full_outer_join.out]: explain plans are notorious and keep changing. Please remove it. Just running the query is sufficient.

Comment (xinzweb) [inline src/test/regress/expected/partition_full_outer_join.out]: Good point. I removed the 'explain'.

File: src/test/regress/sql/partition_full_outer_join.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+-- end_ignore
+
+-- expect GPOPT fall back to legacy query optimizer 
+-- since we don't support partition elimination through full outer joins
+select * from s1 full outer join s2 on s1.d1 = s2.d2 and s1.p1 = s2.p2 where s1.p1 = 1;
+
+-- start_ignore
+drop table if exists s1;
+drop table if exists s2;
+-- end_ignore
```
Comment (vraghavan78) [inline src/test/regress/sql/partition_full_outer_join.sql]: +1

**Comment by hsyuan** [comment]
LGTM.

**Comment by xinzweb** [comment]
Already covered in #265

**Pull request closed without merging**